src/metabase/api/table.clj

```diff
@@ -442,14 +443,18 @@
                                                                         [:= :most_recent true]]
                                                              :order-by [[:id :desc]]
                                                              :limit    1}
-                                                            :id id)
+                                             :id id)
                                            first :status)
-        db (t2/select-one Database :id database_id)]
+        db (t2/select-one Database :id database_id)
+        ;; a native model can have columns with keys as semantic types on if a user specified it
+        trust-semantic-keys? (and (= (:type card) :model)
+                                     (= (-> card :dataset_query :type) :native))]
```

Comment:
A regular (mbql) model can also have metadata overrides with FKs for columns that are not FKs in the DB

Reply:
@ranquild Do you have an example when that information gets lost? (The FK override on mbql models worked for me.)

Reply:
I don't have an example. I think you're right and that's the fix we need to do. It should be safe to "trust" FK metadata on native queries because it won't be set automatically.

Reply:
I've just noticed the typo in the comment. I meant "only if a user specified it", not "on if a user specified it".